Sources/_SwiftUINavigationState/ButtonState.swift

```diff
@@ -41,6 +41,18 @@ public struct ButtonState<Action>: Identifiable {
     self.label = label()
   }
 
+  // TODO: Is this initializer worth it?
+  //       Allows for ButtonState(role: .destructive, send: .confirmDeletion)
+  public init(
+    role: Role? = nil,
+    send action: Action? = nil,
+    label: () -> TextState
+  ) {
+    self.role = role
+    self.action = action.map(ButtonAction.send)
+    self.label = label()
+  }
+
   // TODO: Keep these of leave these?
 //  public init(
 //    _ titleKey: LocalizedStringKey,
```

Comment:
I think that this is required for localizable string extraction from Xcode. But it requires to disfavor the `String` overload too. For example, the following extracts "ToBeLocalized" as expected. Of course, it would apply to all similar situations where the user can pass a string literal.

```swift
import SwiftUI
let wrapped = Wrapper("ToBeLocalized")
struct Wrapper {
  init(_ string: LocalizedStringKey) {}
  
  @_disfavoredOverload
  init(_ string: String) {}
  
  @_disfavoredOverload
  init<S: StringProtocol>(_ string: S) {}
}
```

Reply:
Never mind, I checked this out and there is only a `TextState` endpoint, which should already take care of that. I don't think that these overloads are required then.